lib/csv.rb

```diff
@@ -1179,7 +1179,7 @@ def each(&block)
   def read
     rows = to_a
     headers = parser.headers
-    if headers
+    if headers || parser.use_headers
```

Comment:
Can we remove `headers ||`?

Reply:
Yes. Only parser.use_headers? is enough condition.